src/smc-project/jupyter/jupyter.ts

```diff
@@ -237,13 +237,16 @@ export class JupyterKernel extends EventEmitter
     dbg("spawning kernel...");
 
     const opts: any = { detached: true, stdio: "ignore" };
-    if (this.name.indexOf("sage")) {
-      // special environment for sage-based kernels
+
+    if (this.name.indexOf("sage") == 0) {
```

Comment:
Before I merge this, is this needed?  It seems like this was set before **if and only if** it was NOT a sage kernel, so it was never being set in the case of any sage kernel.... hence it can't be needed, and a better fix is to delete the code?

Reply:
now I understand, well, I think it's good to set the variable for R. the one for python is already set, and well, I'll refine the patch after a few tests.